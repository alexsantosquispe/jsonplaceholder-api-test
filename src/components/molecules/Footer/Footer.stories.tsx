import { Meta, StoryObj } from '@storybook/react';
import { BrowserRouter } from 'react-router-dom';

import { Footer } from './Footer';

const meta = {
  title: 'Molecules/Footer',
  component: Footer,
  parameters: {
    layout: 'centered'
  },
  decorators: [
    (Story) => {
      return (
        <BrowserRouter>
          <div className="flex w-full flex-col rounded-lg border border-gray-200">
            <div className="flex h-[30rem] flex-col items-center justify-center">
              <div>Main Content</div>
            </div>
            <Story />
          </div>
        </BrowserRouter>
      );
    }
  ]
} satisfies Meta<typeof Footer>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {}
};
