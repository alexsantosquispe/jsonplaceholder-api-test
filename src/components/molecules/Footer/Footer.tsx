import { Link } from 'react-router-dom';

export const Footer = () => {
  return (
    <footer className="flex h-14 w-full justify-center border-t border-dashed border-gray-200 dark:border-white/15">
      <div className="flex w-[80rem] items-center justify-center gap-x-2 md:border-x md:border-dashed md:border-gray-200 md:dark:border-white/15">
        <p className="text-primary/65 text-xs font-light md:text-sm dark:text-white/65">
          Built by
          <span className="text-primary dark:text-white"> Alex Santos.</span>
          &nbsp;Source code available on &nbsp;
          <Link
            to="https://github.com/alexsantosquispe/jsonplaceholder-api-test"
            target="_blank"
            rel="noopener noreferrer"
            className="text-primary underline dark:text-white"
          >
            Github
          </Link>
        </p>
      </div>
    </footer>
  );
};
