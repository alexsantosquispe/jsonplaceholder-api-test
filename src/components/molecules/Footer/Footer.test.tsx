import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';

import { Footer } from './Footer';

describe('Footer', () => {
  it('should render the component correctly', () => {
    const component = render(
      <BrowserRouter
        future={{
          v7_startTransition: true,
          v7_relativeSplatPath: true
        }}
      >
        <Footer />
      </BrowserRouter>
    );

    expect(component).toMatchSnapshot();
  });

  it('should display the content', () => {
    render(
      <BrowserRouter
        future={{
          v7_startTransition: true,
          v7_relativeSplatPath: true
        }}
      >
        <Footer />
      </BrowserRouter>
    );

    expect(screen.getByText(/Built by/i)).toBeInTheDocument();

    expect(screen.getByText('Alex Santos.')).toBeInTheDocument();

    expect(screen.getByText(/Source code available on/i)).toBeInTheDocument();

    expect(screen.getByText('Github')).toBeInTheDocument();
  });

  it('should have the correct GitHub link with target and rel attributes', () => {
    render(
      <BrowserRouter
        future={{
          v7_startTransition: true,
          v7_relativeSplatPath: true
        }}
      >
        <Footer />
      </BrowserRouter>
    );

    const githubLink = screen.getByText('Github');

    expect(githubLink).toBeInTheDocument();

    expect(githubLink).toHaveAttribute(
      'href',
      'https://github.com/alexsantosquispe/jsonplaceholder-api-test'
    );

    expect(githubLink).toHaveAttribute('target', '_blank');

    expect(githubLink).toHaveAttribute('rel', 'noopener noreferrer');
  });
});
