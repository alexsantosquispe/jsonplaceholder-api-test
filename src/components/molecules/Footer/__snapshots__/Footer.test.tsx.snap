// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render the component correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="flex h-14 w-full justify-center border-t border-dashed border-gray-200 dark:border-white/15"
      >
        <div
          class="flex w-[80rem] items-center justify-center gap-x-2 md:border-x md:border-dashed md:border-gray-200 md:dark:border-white/15"
        >
          <p
            class="text-primary/65 text-xs font-light md:text-sm dark:text-white/65"
          >
            Built by
            <span
              class="text-primary dark:text-white"
            >
               Alex Santos.
            </span>
             Source code available on  
            <a
              class="text-primary underline dark:text-white"
              href="https://github.com/alexsantosquispe/jsonplaceholder-api-test"
              rel="noopener noreferrer"
              target="_blank"
            >
              Github
            </a>
          </p>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="flex h-14 w-full justify-center border-t border-dashed border-gray-200 dark:border-white/15"
    >
      <div
        class="flex w-[80rem] items-center justify-center gap-x-2 md:border-x md:border-dashed md:border-gray-200 md:dark:border-white/15"
      >
        <p
          class="text-primary/65 text-xs font-light md:text-sm dark:text-white/65"
        >
          Built by
          <span
            class="text-primary dark:text-white"
          >
             Alex Santos.
          </span>
           Source code available on  
          <a
            class="text-primary underline dark:text-white"
            href="https://github.com/alexsantosquispe/jsonplaceholder-api-test"
            rel="noopener noreferrer"
            target="_blank"
          >
            Github
          </a>
        </p>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
